File: backend/src/main/java/com/hospital/service/AppointmentService.java
package com.hospital.service;

import com.hospital.entity.Appointment;
import java.util.List;

public interface AppointmentService {
    Appointment createAppointment(Appointment appointment);
    Appointment getAppointmentById(Long id);
    List<Appointment> getAllAppointments();
    Appointment updateAppointment(Long id, Appointment appointmentDetails);
    void deleteAppointment(Long id);
    List<Appointment> getAppointmentsByPatientId(Long patientId);
    List<Appointment> getAppointmentsByDoctorId(Long doctorId);
}

File: frontend/src/services/appointmentService.js
import api from './api';

export const getAllAppointments = () => api.get('/appointments');
export const getAppointmentById = (id) => api.get(`/appointments/${id}`);
export const createAppointment = (appointment) => api.post('/appointments', appointment);
export const updateAppointment = (id, appointment) => api.put(`/appointments/${id}`, appointment);
export const deleteAppointment = (id) => api.delete(`/appointments/${id}`);
export const getAppointmentsByPatientId = (id) => api.get(`/appointments/patient/${id}`);
export const getAppointmentsByDoctorId = (id) => api.get(`/appointments/doctor/${id}`);
